tests/integration_tests/tests/agentless_tests/test_runtime_functions.py

```diff
@@ -0,0 +1,197 @@
+########
+# Copyright (c) 2019 Cloudify Platform Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+import os
+
+from integration_tests import AgentlessTestCase
+from integration_tests.tests.utils import get_resource as resource
+
+
+class TestRuntimeFunctionEvaluation(AgentlessTestCase):
+    """Tests for the runtime_only_evaluation feature.
+
+    Those tests will run a workflow which will gather values from various
+    sources (node properties, operation inputs) into runtime properties,
+    which allows us to do assertions on the runtime properties.
+    """
+    BASE_VALUE = 'aaa'
+    CHANGED_VALUE = 'bbb'
+
+    def setUp(self):
+        super(TestRuntimeFunctionEvaluation, self).setUp()
+        self.bp_path = resource('dsl/deployment_update_functions.yaml')
+
+    def _assert_properties(self, deployment, value):
+        instances_node1 = self.client.node_instances.list(
+            node_id='node1', deployment_id=deployment.id)
+        self.assertEqual(len(instances_node1), 1)
+        instances_node2 = self.client.node_instances.list(
+            node_id='node2', deployment_id=deployment.id)
+        ni1 = instances_node1[0]
+        ni2 = instances_node2[0]
+
+        outputs = self.client.deployments.outputs.get(deployment.id)['outputs']
+        self.assertEqual(outputs['out1'], value)
+        self.assertEqual(outputs['out2'], value)
+        self.assertEqual(outputs['out3'], value)
+        self.assertEqual(ni1.runtime_properties['arg_value'], value)
+        self.assertEqual(ni1.runtime_properties['prop1_value'], value)
+        self.assertEqual(ni1.runtime_properties['prop2_value'], value)
+        self.assertEqual(ni1.runtime_properties['source_input_value'], value)
+        self.assertEqual(ni1.runtime_properties['target_input_value'], value)
+        self.assertEqual(ni2.runtime_properties['source_input_value'], value)
+        self.assertEqual(ni2.runtime_properties['target_input_value'], value)
+        self.assertEqual(ni1.runtime_properties['source_prop_value'], value)
+        self.assertEqual(ni2.runtime_properties['target_prop_value'], value)
+
+    def _manual_deployment_update(self, deployment):
```

Comment:
so this is done like this for saving test time?

Reply:
this is pretty much mocking out deployment update (if you wanna be technical it's more of a fake than a mock i guess...)
to check that the runtime-functions stuff is independent of deployment updates (because maybe all the work was done by the update?)

Reply:
added a note about this to docstring